scripts/all_configs.go

```diff
@@ -88,31 +93,53 @@ func fetchAndWriteConfig(uic *uiClient, project string) {
 }
 
 func main() {
-	opts := &cli.Options{}
-	var parser = flags.NewParser(opts, flags.Default)
-	_, err := parser.Parse()
-	if err != nil {
-		panic(err)
-	}
-	settings, err := cli.LoadSettings(opts)
+	app := cli.NewApp()
+	app.Name = "all_configs"
+	app.Usage = "Evergreen Configuration Download Tool"
+
+	userHome, err := homedir.Dir()
 	if err != nil {
-		panic(err)
+		// workaround for cygwin if we're on windows but couldn't get a homedir
+		if runtime.GOOS == "windows" && len(os.Getenv("HOME")) > 0 {
+			userHome = os.Getenv("HOME")
+		}
 	}
-	uic := &uiClient{
-		client: &http.Client{},
-		user:   settings.User,
-		key:    settings.APIKey,
-		uiRoot: settings.UIServerHost,
+	confPath := filepath.Join(userHome, ".evergreen.yml")
+
+	// These are global options. Use this to configure logging or
+	// other options independent from specific sub commands.
+	app.Flags = []cli.Flag{
+		cli.StringFlag{
+			Name:  "conf, config, c",
+			Usage: "specify the path for the evergreen CLI config",
+			Value: confPath,
+		},
 	}
+	app.Action = func(c *cli.Context) error {
+		settings, err := operations.NewClientSetttings(c.GlobalString("config"))
+		if err != nil {
+			return err
+		}
 
-	// fetch list of all projects
-	resp := uic.get("projects")
-	pList := struct {
-		Projects []string `json:"projects"`
-	}{}
-	mustDecodeJSON(resp.Body, &pList)
+		uic := &uiClient{
+			client: &http.Client{},
```

Comment:
use the pooled client?

Reply:
leaving this alone in favour of merging it into the binary